#!/usr/bin/env python
# Copied from franka_ros/franka_example_controllers/cfg/compliance_param.py
PACKAGE = "frankpiv_controller"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("translational_stiffness", double_t, 0, "Cartesian translational stiffness in N/m", 400, 0, 400)
gen.add("rotational_stiffness", double_t, 0, "Cartesian rotational stiffness in Nm/rad", 30, 0, 30)
gen.add("nullspace_stiffness", double_t, 0, "Stiffness of the joint space nullspace controller (the desired configuration is the one at startup)", 0.5, 0, 100)
gen.add("error_scaling_z", double_t, 0, "Scaling of the error in z direction", 0.5, 0.1, 10)
gen.add("error_scaling_xy", double_t, 0, "Scaling of the error in xy direction", 1.0, 0.1, 10)

exit(gen.generate(PACKAGE, "dynamic_compliance", "compliance_param"))
